action.yml

```diff
@@ -29,25 +29,16 @@ runs:
       with:
         fetch-depth: 0
 
-    - name: Setup Go
-      uses: actions/setup-go@v4
-      with:
-        go-version: "1.22.5"
-
-    - name: Install pizza CLI
-      run: |
-        go install github.com/open-sauced/pizza-cli@${{ inputs.cli-version }}
-      shell: bash
-
     - name: Check pizza CLI version
-      run: pizza-cli version
+      run: |
+        DOCKERIMAGE="ghcr.io/open-sauced/pizza-cli:${{ inputs.cli-version }}"
+        docker run $DOCKERIMAGE version
       shell: bash
 
     - name: Run pizza CLI command
       run: |
-        pwd
-        ls -la
-        pizza-cli ${{ inputs.pizza-args }}
+        DOCKERIMAGE="ghcr.io/open-sauced/pizza-cli:${{ inputs.cli-version }}"
+        docker run -v ./:/repository $DOCKERIMAGE ${{ inputs.pizza-args }}
```

Comment:
We shouldn't invoke docker directly on the GitHub runner.

GitHub offers a mechanism to run container jobs via `image` and other abstractions to specifiy the container runtime:

```yaml
- name: Run pizza CLI command
  runs-on: ubuntu-latest
  container:
    image: ghcr.io/open-suaced/pizza-cli:${{ inputs.cli-version }}
    volumes:
      - ./:/repository
  steps:
    - run: pizza ${{ inputs.pizza-args }}
```

https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container

Reply:
That syntax doesn't appear to work in a, action.yml with a composite action. I'll see what else I need to adjust to use jobs with images as you suggest.

Reply:
Probably not worth going down this rabbit hole: go ahead and ship it!